llama_index/schema.py

```diff
@@ -432,6 +434,7 @@ def get_metadata_str(self, mode: MetadataMode = MetadataMode.ALL) -> str:
     def set_content(self, value: str) -> None:
         """Set the content of the node."""
         self.text = value
+        self._ensure_hash()
```

Comment:
So, I guess this only solves the issue when `set_content()` is used. But if the text attribute is modified directly, `node.text = "new"`, then it still doesn't update. This also misses the metadata as well

I think a better option here is to keep the old root validator, but add this to the TextNode object
```
class  Config:
    validate_assignment = True
```

Now with that config set inside the text node, this works

```
>>> node = TextNode(text="test")
>>> node.hash
'6c51d46392bb993df662da884185f4252c581f592c7e35024e7b12f4bdaad075'
>>> node.text = "123"
>>> node.hash
'018e6bd0a7445808763fb04f99926b055cc58fb3fab9a61be01605bf6d663486'
```

Reply:
Makes sense, thanks. I updated the pr as you suggested

Reply:
awesome, thanks!